### PR TITLE
Update to use new chaincode tests

### PR DESCRIPTION
**Pull request by mbwhite**
Signed-off-by: Matthew B White <whitemat@uk.ibm.com>

### CODE REVIEW

**Review by jt-nti — CHANGES_REQUESTED** [review]
There seem to be a few tmp files, which I guess shouldn't be checked in. The tmp files seem to be related to modifying go.mod files in place, which could also ten get checked in inadvertently. I think the tests also require the fabric-samples repo and fabric binaries for the test network. All of which would be better/cleaner if the tests created a temporary directory outside the repo to run everything in. Something like, create temp directory, copy required files/chaincode from the repo modifying go.mod files on the way, clone fabric-samples into temp directory and put the fabric binaries in the right place, etc. etc.

File: .azure-pipelines/scripts/fabric-quickly.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ROOTDIR=$(cd "$(dirname "$0")" && pwd)
+
+prefix=fab-test-net
+tempdir=$(mktemp -d -t "$prefix.XXXXX") || error_exit "Error creating temporary directory"
+
+DIR=${FABRIC_DIR:="${ROOTDIR}/fabric"}
+
+if [ -d "$DIR" ]
+then
+	if [ "$(ls -A $DIR)" ]; then
```
[the remark this answers is not in the document]
Comment (mbwhite) [reply .azure-pipelines/scripts/fabric-quickly.sh:16]: oh no... yes fixes

File: integrationtest/features/feature-1-basic-checks.feature [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Copyright the Hyperledger Fabric contributors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (mbwhite) [reply integrationtest/features/feature-1-basic-checks.feature:1]: yes.. they are now removed

**Comment by mbwhite** [comment]
thanks @jt-nti 

On the temporary directory, currently the fabric-samples are extracted into a temp directory, pruned, and then copied locally. so it's just the test-network and binaries. 

Will think some more what the cleanest way to do this, as the location of the temporary directory will need to be carried through the various steps of the build....

**Comment by ryjones** [comment]
@mbwhite I cherry-picked this to the tip of main

File: ci/scripts/setup-integration-chaincode.sh [hunk 3]
```diff
@@ -8,20 +8,21 @@
...
 
+CHAINCODE_DIR=$ROOTDIR/../../integrationtest/chaincode
+ls -lart $CHAINCODE_DIR
+pushd $CHAINCODE_DIR
 for testCC in */; do
-    cd $testCC
-    cat go.mod > tmp.go.mod
-    echo 'replace github.com/hyperledger/fabric-contract-api-go => ../../../..' >> go.mod
+    pushd $testCC
+    # cat go.mod > tmp.go.mod
```
Comment (jt-nti) [inline ci/scripts/setup-integration-chaincode.sh:19]: There's some commentated out code in this script which it would be good to tidy up

**Review by jt-nti — CHANGES_REQUESTED** [review]
Looks good- just some commented out code and a fabric-chaincode-integration-0.5.0.tgz tarball which it would be good to clear up

**Comment by jt-nti** [comment]
The new test tool is ready! 🎉
https://github.com/hyperledger/fabric-test/packages/1545024

File: ci/scripts/setup-integration-chaincode.sh [hunk 4]
```diff
@@ -8,20 +8,19 @@
 # by the integration tool which doesn't have the local files. Removes the replace to stop the integration tests
 # breaking as the dot path doesn't exist there.
 
-DIR=$(pwd)
-CHAINCODE_DIR=$DIR/.azure-pipelines/resources/chaincode
+# Note the actually committed go.mod files of the conahincode already have the
```
Comment (jt-nti) [inline ci/scripts/setup-integration-chaincode.sh:11]: Typo

**Review by jt-nti — APPROVED** [review]
Just a minor typo left but seems good otherwise

**Review by jt-nti — APPROVED** [review]

**Pull request merged**